### DISCUSSION TITLE
hyper_util::client::legacy::Client request method with Request<Full<Bytes>>

### DISCUSSION THREAD

**Discussion by Vagelis-Prokopiou** [hyperium/hyper#3555] [Q&A] [answered]
I have a rustls hyper client defined like this:
```
...
let client: Client<_, Full<Bytes>> = Client::builder(hyper_util::rt::TokioExecutor::new()).build(https);
```

When I try to execute a request of type `Request<Full<Bytes>>` I get the following error:
```
error[E0308]: mismatched types
   --> src\handlers.rs:353:18
    |
353 |         .request(auth_request)
    |          ------- ^^^^^^^^^^^^ expected `Request<Incoming>`, found `Request<Full<Bytes>>`
    |          |
    |          arguments to this method are incorrect
    |
    = note: expected struct `hyper::Request<hyper::body::Incoming>`
               found struct `hyper::Request<http_body_util::Full<hyper::body::Bytes>>`
note: method defined here
   --> ...\.cargo\registry\src\index.crates.io-6f17d22bba15001f\hyper-util-0.1.3\src\client\legacy\client.rs:201:12
    |
201 |     pub fn request(&self, mut req: Request<B>) -> ResponseFuture {
```

I did not understand how to fix this. I cannot directly instantiate the `Incoming` struct, and I would suppose that `Full<Bytes>` is compatible already.

Any ideas?

**Comment by seanmonstar** [comment, marked as answer]
It's hard to know from just that message. It looks like whatever is being used at L353 expects a different body type than what `auth_request` is.

Is `auth_request` coming from a hyper server? Is it `Request<Incoming>`? If so, you'd need to make the body types the same somehow. Either by collecting the whole body into a single full buffer, or changing your client body type to a dynamic one, like `BoxBody`.

**Comment by Vagelis-Prokopiou** [comment]
Yes. It's Request<Incoming>.
I will try the BoxBody and let you know.
Thanks a lot for now.

**Comment by Vagelis-Prokopiou** [comment]
I decided to create 2 clients.
One for bodies of type `Incoming` and one for bodies of type `Full<Bytes>`.
I think it is more efficient during runtime because I avoid the aggregation of the bodies bytes and also the boxing.
I did not benchmark anything though. Not that necessary.
